mongo/datadog_checks/mongo/collectors/replica.py

```diff
@@ -0,0 +1,104 @@
+import time
+
+from six.moves.urllib.parse import urlsplit
+
+from datadog_checks.mongo.collectors.base import MongoCollector
+from datadog_checks.mongo.common import REPLSET_MEMBER_STATES, SOURCE_TYPE_NAME, get_state_name
+
+try:
+    import datadog_agent
+except ImportError:
+    from datadog_checks.base.stubs import datadog_agent
+
+
+class ReplicaCollector(MongoCollector):
+    """Collect replica set metrics by running the replSetGetStatus command. Also keep track of the previous node state
+    in order to submit events on any status change.
+    """
+
+    def __init__(self, check, tags, last_state):
+        super(ReplicaCollector, self).__init__(check, "admin", tags)
+        # Members' last replica set states
+        self._last_state = last_state
+
+        # Makes a reasonable hostname for a replset membership event to mention.
+        uri = urlsplit(self.check.clean_server_name)
+        if '@' in uri.netloc:
+            self.hostname = uri.netloc.split('@')[1].split(':')[0]
+        else:
+            self.hostname = uri.netloc.split(':')[0]
+        if self.hostname == 'localhost':
+            self.hostname = datadog_agent.get_hostname()
```

Comment:
(Nit) Seems quite low-level, maybe this could become a small `make_event_friendly_replset_hostname()` helper in this module, and then we'd use it like this…?

```suggestion
        self.hostname = make_event_friendly_replset_hostname(self.check.clean_server_name)
```